load-config.js

```diff
@@ -1,35 +1,52 @@
-let config = {};
+let userConfig;
 let fs = require('fs');
 let { join } = require('path');
 let root = fs.existsSync('/data') ? '/data' : __dirname;
 
 try {
   // @ts-expect-error TS2307: we expect this file may not exist
-  config = require('./config');
+  userConfig = require('./config');
 } catch (e) {
   // do nothing
 }
 
+/** @type {Omit<import('./config-types').Config, 'mode'>} */
+let defaultConfig = {
+  port: 5006,
+  hostname: '::',
+  serverFiles: join(root, 'server-files'),
+  userFiles: join(root, 'user-files'),
+  webRoot: join(__dirname, 'node_modules', '@actual-app', 'web', 'build')
+};
+
+/** @type {import('./config-types').Config} */
+let config;
 if (process.env.NODE_ENV === 'test') {
   config = {
     mode: 'test',
-    port: 5006,
-    hostname: '::',
-    serverFiles: join(__dirname, 'test-server-files'),
-    userFiles: join(__dirname, 'test-user-files')
+    ...defaultConfig
   };
 } else {
   config = {
     mode: 'development',
-    port: 5006,
-    hostname: '::',
-    serverFiles: join(root, 'server-files'),
-    userFiles: join(root, 'user-files'),
-    ...config
+    ...defaultConfig,
+    ...userConfig
   };
 }
 
-// The env variable always takes precedence
-config.userFiles = process.env.ACTUAL_USER_FILES || config.userFiles;
-
-module.exports = config;
+module.exports = {
+  ...config,
+  port: +process.env.ACTUAL_PORT || +process.env.PORT || config.port,
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review (if it exists :^)  -->
**👏 praise:** ‏good idea to add these as env vars. Should we also use `dotenv` to pick automatically pick these up from `.env` file? And then we could also create a `.env.example` file where we list them all out and provide a basic documentation. WDYT?

Reply:
IMO it’s more ergonomic to use the `config.json` if you can, this is intended for Docker and such where it’s easy to set env vars but hard to inject a specific file. (For example in my other PR I use it to override the web root for the `edge` images).